### PR TITLE
HRIS-198-1 [FE] Bugfix on new day always starts on monday

### PR DESCRIPTION
**Pull request by JohnRoyCabezas**
## Issue Link

[Backlog Link](https://framgiaph.backlog.com/view/HRIS-198#comment-144744821)

## Definition of Done

- [x] Users can no longer face the bug/issue, where creating a new day always starts on a Monday.

## Pre-condition

run docker
go to http://localhost:3000/schedule-management

## Expected Output

It should fix the issue/bug, where creating a new day always starts on a Monday.

## Screenshots/Recordings

Proof

[screen-capture.webm](https://user-images.githubusercontent.com/109579325/232644912-9ee4c532-9699-44ad-876c-3cd2cdc8c593.webm)

### CODE REVIEW

File: client/src/pages/schedule-management.tsx [hunk 1]
```diff
@@ -66,27 +66,43 @@ const ScheduleManagement: NextPage = (): JSX.Element => {
...
-    setValue('friday.timeIn', EmployeeSchedule?.days[4].timeIn as string)
-    setValue('friday.timeOut', EmployeeSchedule?.days[4].timeOut as string)
-    setValue('saturdaySelected', EmployeeSchedule?.days[5].isDaySelected as boolean)
-    setValue('saturday.timeIn', EmployeeSchedule?.days[5].timeIn as string)
-    setValue('saturday.timeOut', EmployeeSchedule?.days[5].timeOut as string)
-    setValue('sundaySelected', EmployeeSchedule?.days[6].isDaySelected as boolean)
-    setValue('sunday.timeIn', EmployeeSchedule?.days[6].timeIn as string)
-    setValue('sunday.timeOut', EmployeeSchedule?.days[6].timeOut as string)
+    for (let x = 0; x < 7; x++) {
+      if (EmployeeSchedule?.days[x].workingDay === 'Monday') {
```
Comment (arden-dave) [inline client/src/pages/schedule-management.tsx:70]: You might want to make strings like Monday, Tuesday, etc ... constants and import them

**Review by arden-dave — CHANGES_REQUESTED** [review]

File: client/src/pages/schedule-management.tsx (hunk 1 above)
Comment (JohnRoyCabezas) [reply client/src/pages/schedule-management.tsx]: Updated. Thanks!

**Review by arden-dave — APPROVED** [review]
Approve. 👍

File: client/src/pages/schedule-management.tsx [hunk 2]
```diff
@@ -66,27 +67,43 @@ const ScheduleManagement: NextPage = (): JSX.Element => {
...
+        setValue('saturday.timeIn', EmployeeSchedule?.days[x].timeIn)
+        setValue('saturday.timeOut', EmployeeSchedule?.days[x].timeOut)
+      }
+      if (EmployeeSchedule?.days[x].workingDay === WeekDays.SUNDAY) {
+        setValue('sundaySelected', EmployeeSchedule?.days[x].isDaySelected)
+        setValue('sunday.timeIn', EmployeeSchedule?.days[x].timeIn)
+        setValue('sunday.timeOut', EmployeeSchedule?.days[x].timeOut)
+      }
+    }
   }, [id, isLoading])
```
Comment (abduljalilpalala) [inline client/src/pages/schedule-management.tsx:107]: Put this in a separate function and try this code.

```
const dayProperties = {
  MONDAY: { selected: 'mondaySelected', timeIn: 'monday.timeIn', timeOut: 'monday.timeOut' },
  TUESDAY: { selected: 'tuesdaySelected', timeIn: 'tuesday.timeIn', timeOut: 'tuesday.timeOut' },
  WEDNESDAY: { selected: 'wednesdaySelected', timeIn: 'wednesday.timeIn', timeOut: 'wednesday.timeOut' },
  THURSDAY: { selected: 'thursdaySelected', timeIn: 'thursday.timeIn', timeOut: 'thursday.timeOut' },
  FRIDAY: { selected: 'fridaySelected', timeIn: 'friday.timeIn', timeOut: 'friday.timeOut' },
  SATURDAY: { selected: 'saturdaySelected', timeIn: 'saturday.timeIn', timeOut: 'saturday.timeOut' },
  SUNDAY: { selected: 'sundaySelected', timeIn: 'sunday.timeIn', timeOut: 'sunday.timeOut' },
};

for (const day of EmployeeSchedule?.days) {
  const { workingDay, isDaySelected, timeIn, timeOut } = day;
  if (workingDay in dayProperties) {
    const { selected, timeIn: dayTimeIn, timeOut: dayTimeOut } = dayProperties[workingDay];
    setValue(selected, isDaySelected);
    setValue(dayTimeIn, timeIn);
    setValue(dayTimeOut, timeOut);
  }
}
```

**Pull request merged**